### ISSUE TITLE
apply new cpuminer (yespower)

### ISSUE THREAD

**Issue by cryptozeny** [OttyLab/BitZenyAndroidMiningLibrary#34]
https://github.com/cryptozeny/cpuminer-mc-yespower

hi i released a new cpuminer using yespower. it boosts 2~15% hashpower. it has an improve on smartphones! :smiley: it based on macchky's one.

ryzen + 2%
intel + 3%~7%
arm-aarch64 (smartphones) + 15%
arm-v7l (neon): not supported yet by algorithm author, and it can run as fallback mode (no boost). hopefully he said it will come with next release.

**Comment by cryptozeny** [comment]
today i pushed yespower to official cpuminer what you currently using. you can apply with 

`-a yespower` option 😉

**Comment by OttyLab** [comment]
Thanks, I'll check that.

**Comment by OttyLab** [comment]
@cryptozeny 

support-yespower branch supports yespower. Could you try this out? (iOS also supports)

**Comment by cryptozeny** [comment]
@OttyLab i lost my develop environments for android, so it takes some time... :sob: could you give me a test release?

**Comment by OttyLab** [comment]
@cryptozeny 

Here's the build. (Sorry, the signature is changed because I lost the password 😅 )
[build.zip](https://github.com/OttyLab/BitZenyAndroidMiningLibrary/files/2238608/build.zip)

In my environment, yespower is worse than yescrypt. Is any build parameter required for optimization?

**Comment by cryptozeny** [comment]
it wont install on my note8 😅 i will try an another device

**Comment by cryptozeny** [comment]
@OttyLab please use this build option? it makes faster

```
./configure CFLAGS="-O2 -fomit-frame-pointer"
```

**Comment by OttyLab** [comment]
@cryptozeny 

> it wont install on my note8

What kind of error did you get? If the device has previous version, the new one fails due to the signature mismatch. In that case, uninstalling the previous one would work.


>  -fomit-frame-pointer

It improves the hash power, but looks not enough. I'll see to it.

**Comment by OttyLab** [comment]
Hmm, it's weird...

Nexus 7:

|.      | yescript    | yespower   |
|:----|:------------|:------------|
| (1)  | 0.066 kh/s | 0.071 kh/s |
| (2) | 0.12 kh/s   |  0.1 kh/s    |

(1): run minerd on adb shell
(2): run libmier on the app

The hash rate is way different on yescript between (1) and (2)

**Comment by cryptozeny** [comment]
wired...🤔

**Comment by OttyLab** [comment]
I've checked the code but I could not find out any problem.

`libmienr.c` (used by Android app) is created based on `cpu-miner.c` to support start/stop the threads. That means the differences in `libminer.c` should not affect the hash rate in theory.

I performed additional tests with cpuminer2 on Nexus7

1. on adb shell
+ minerd: 0.10kHash/s
+ libminer: 0.10kHash/s

2. on Android app
+ minerd: cannot test (does not support start/stop etc)
+ libminer: 0.13kHash/s

So, it seems that the hash rate improves when `libminer` is called from Android app for some reason, but I do not know what causes this result.
